### PR TITLE
Remove usage of a SecurityManager from EE11 

### PR DESCRIPTION
**Pull request by gregw**
Remove usage of a SecurityManager from EE11 
and disable by default all other usages.

### CODE REVIEW

File: jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ServletContextHandler.java [hunk 1]
```diff
@@ -324,16 +322,17 @@ public void dump(Appendable out, String indent) throws IOException
             new DumpableCollection("initparams " + this, getInitParams().entrySet()));
     }
 
+    @Deprecated(forRemoval = true, since = "12.0.0")
```
Comment (janbartel) [inline jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ServletContextHandler.java]: Not really 12.0.0? More like 12.0.12?

File: jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ServletContextHandler.java [hunk 2]
```diff
@@ -324,16 +322,17 @@ public void dump(Appendable out, String indent) throws IOException
...
     public boolean isUsingSecurityManager()
     {
-        return _usingSecurityManager;
+        return false;
     }
 
+    @Deprecated(forRemoval = true, since = "12.0.0")
```
Comment (janbartel) [inline jetty-ee11/jetty-ee11-servlet/src/main/java/org/eclipse/jetty/ee11/servlet/ServletContextHandler.java]: See above

**Review by janbartel — CHANGES_REQUESTED** [review]
Do we need to deprecate or otherwise get rid of `PrivilegedThreadFactory`?

**Comment by gregw** [comment]
> Do we need to deprecate or otherwise get rid of `PrivilegedThreadFactory`?

It is package scope already, so an internal class, so we can get rid of it once there are no more JVMs with SecurityManagers without deprecation I think

**Review by janbartel — APPROVED** [review]

**Pull request merged**